segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -648,12 +651,19 @@ public void updateTableEntries(final WireCommands.UpdateTableEntries updateTable
         }
 
         val timer = new Timer();
-        tableStore.put(segment, entries, TIMEOUT)
-                .thenAccept(versions -> {
-                    connection.send(new WireCommands.TableEntriesUpdated(updateTableEntries.getRequestId(), versions));
-                    this.tableStatsRecorder.updateEntries(updateTableEntries.getSegment(), entries.size(), conditional.get(), timer.getElapsed());
-                })
-                .exceptionally(e -> handleException(updateTableEntries.getRequestId(), segment, operation, e));
+        long tableSegmentOffset = updateTableEntries.getTableSegmentOffset();
+        segmentInfo.thenCompose(properties -> {
+            if (tableSegmentOffset == NULL_TABLE_SEGMENT_OFFSET || tableSegmentOffset == properties.getLength()) {
+                return tableStore.put(segment, entries, TIMEOUT);
```

Comment:
This offset check needs to be preformed atomically with respect to this put.
As is, this code will only check the length at the time the request arrived, but it could change by the time `put` is processed. So the value needs to be passed into tablestore to make the update atomic.

Reply:
@co-jo You can push this check down all the way into `ContainerTableExtensionImpl` in the `commit` method. That method invokes `DirectSegmentAccess.append` which should have an overload that takes an offset (if it doesn't then we should add one (the plumbing underneath is already done)). That will guarantee this to be an atomic length-conditional append.